addons/event/models/event_mail.py

```diff
@@ -44,16 +48,27 @@ class EventTypeMail(models.Model):
         ('before_event', 'Before the event'),
         ('after_event', 'After the event')],
         string='Trigger', default="before_event", required=True)
-    template_id = fields.Many2one(
-        'mail.template', string='Email Template',
-        domain=[('model', '=', 'event.registration')], ondelete='restrict',
-        help='This field contains the template of the mail that will be automatically sent')
+    template_model_id = fields.Many2one('ir.model', string='Template Model', compute='_compute_template_model_id')
+    template_ref = fields.Reference(string='Template', selection='_selection_template_model', required=True, ondelete='cascade')
 
-    @api.model
-    def _get_event_mail_fields_whitelist(self):
-        """ Whitelist of fields that are copied from event_type_mail_ids to event_mail_ids when
-        changing the event_type_id field of event.event """
-        return ['notification_type', 'template_id', 'interval_nbr', 'interval_unit', 'interval_type']
+    @api.depends('notification_type')
+    def _compute_template_model_id(self):
+        mail_model = self.env['ir.model']._get('mail.template')
+        for mail in self:
+            if not mail.notification_type:
```

Comment:
Maybe simply ``mail.tempplate_model_id = mail_model if mail.notification_type == 'mail' else False`` ?

Reply:
it's not really the same condition, it means that if we have "sms" as notification type, we will write 2 times on the same field

Reply:
Well caller should not call super for already-computed fields. It would be something like
1/ mail or False
2/ sms or (self - done).super()
3/ push or (self - done).super()

Reply:
Ah ok I see :)